tests/test_callables.py

```diff
@@ -0,0 +1,9 @@
+from typing import TYPE_CHECKING
+
+from basedtyping.callables import Function
+from basedtyping.type_checks import assert_type
+
+if TYPE_CHECKING:
+
+    def test_function_type() -> None:
+        assert_type[Function](lambda: ...)
```

Comment:
Would be good to also check other types of functions such as `builtin_function` and the various wrappers. I think they are all in `types`.